projects/plugins/crm/src/automation/data-types/class-data-type-invoice.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * Invoice Data Type.
+ *
+ * @package automattic/jetpack-crm
+ * @since $$next-version$$
+ */
+
+namespace Automattic\Jetpack\CRM\Automation\Data_Types;
+
+/**
+ * Invoice Data Type
+ *
+ * @since $$next-version$$
+ */
+class Data_Type_Invoice extends Data_Type_Base {
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public static function get_slug(): string {
+		return 'invoice';
+	}
+
+	/**
+	 * Constructor.
+	 *
+	 * We process the entity data before passing it to validation.
+	 * You can learn more in the "unify_invoice_data" method.
+	 *
+	 * @see Data_Type_Invoice::unify_data()
+	 *
+	 * @param mixed $entity The invoice entity data.
+	 * @return void
+	 *
+	 * @throws \Automattic\Jetpack\CRM\Automation\Data_Type_Exception If the entity is not valid.
+	 */
+	public function __construct( $entity ) {
+		$entity = $this->unify_data( $entity );
+
+		parent::__construct( $entity );
```

Comment:
It seems that the call to the parent constructor should be before unify_data(). The Data_Type_Base constructor performs some data validations.

Reply:
All good. The data validation should happen after `unify_data`